### PR TITLE
Optimized jobs request

### PR DESCRIPTION
**Pull request by nmanovic**

### CODE REVIEW

**Comment by nmanovic** [comment]
@SpecLad , could you please look?

File: cvat/apps/engine/serializers.py [hunk 1]
```diff
@@ -156,10 +150,36 @@ class TasksSummarySerializer(_CollectionSummarySerializer):
 class CommentsSummarySerializer(_CollectionSummarySerializer):
     pass
 
+class BasicSummarySerializer(serializers.Serializer):
+    url = serializers.URLField(read_only=True)
+    count = serializers.IntegerField(read_only=True)
 
-class IssuesSummarySerializer(_CollectionSummarySerializer):
-    pass
+    def to_representation(self, instance):
```
Comment (SpecLad) [inline cvat/apps/engine/serializers.py:157]: I don't think you need to override `to_representation` here. You can replace the fields with `SerializerMethodField`s, and that should cause the methods to be invoked automatically.

File: cvat/apps/engine/serializers.py [hunk 2]
```diff
@@ -156,10 +150,36 @@ class TasksSummarySerializer(_CollectionSummarySerializer):
...
+        }
+
+class LabelsSummarySerializer(BasicSummarySerializer):
+    def get_url(self, request, instance):
+        filter_key = instance.__class__.__name__.lower() + '_id'
+        return reverse('label-list', request=request,
+            query_params={ filter_key: instance.id })
+
+    def get_count(self, instance):
+        return getattr(instance, 'task_labels_count', 0) + getattr(instance, 'proj_labels_count', 0)
```
Comment (SpecLad) [inline cvat/apps/engine/serializers.py:174]: Under what conditions will the attributes be missing?

File: cvat/apps/engine/serializers.py [hunk 3]
```diff
@@ -1059,15 +1083,14 @@ class ProjectReadSerializer(serializers.ModelSerializer):
...
+    labels = LabelsSummarySerializer(source='*')
 
     class Meta:
         model = models.Project
         fields = ('url', 'id', 'name', 'owner', 'assignee',
             'bug_tracker', 'task_subsets', 'created_date', 'updated_date', 'status',
             'dimension', 'organization', 'target_storage', 'source_storage',
-            'tasks', 'labels',
-        )
+            'tasks', 'labels')
```
Comment (SpecLad) [inline cvat/apps/engine/serializers.py:1093]: Unnecessary change.

File: cvat/apps/engine/views.py [hunk 4]
```diff
@@ -1234,13 +1242,12 @@ def preview(self, request, pk):
...
-        'segment__task__project__label_set__sublabels__attributespec_set',
-        'segment__task__label_set__attributespec_set',
-        'segment__task__project__label_set__attributespec_set'
-    ).all()
+    queryset = Job.objects.select_related('assignee', 'segment__task__data',
+        'segment__task__project').annotate(Count('issues', distinct=True),
+        task_labels_count=Count('segment__task__label',
+            filter=Q(segment__task__label__parent__isnull=True), distinct=True),
+        proj_labels_count=Count('segment__task__project__label',
+            filter=Q(segment__task__project__label__parent__isnull=True), distinct=True)).all()
```
Comment (SpecLad) [inline cvat/apps/engine/views.py:1250]: Could you reformat this like in `ProjectViewSet`? It's hard to understand which arguments are for what.

File: cvat/apps/engine/serializers.py (hunk 1 above)
Comment (nmanovic) [reply cvat/apps/engine/serializers.py:157]: I like the proposal but the method has common logic. It handles the case when the request is None. I will prefer to leave it as is. Otherwise, I have to repeat it 4 times.
